packages/modal/src/Modal.js

```diff
@@ -0,0 +1,226 @@
+    `
+    opacity: 0;
+  `}
+`
+
+const Dialog = styled(DialogContent)`
+  ${color} 
+  ${width}
+  ${height}
+  max-width: calc(100vw - 32px);
+  position: relative;
+  margin-left: auto;
+  margin-right: auto;
+  box-shadow: ${props => props.theme.boxShadows[3]};
+  &:focus {
+    outline: none;
+  }
+  max-height: 100vh;
+  ${props =>
+    props.enableoverflow &&
+    `
+    max-height: none;
+  `}
+
+  transform: scale(0.5);
+  transition: transform .5s cubic-bezier(0.50, 0.00, 0.25, 1.00);
+
+  ${props =>
+    props.transitionstate === 'entering' &&
+    `
+    transform: scale(0.5);
+  `}
+
+  ${props =>
+    props.transitionstate === 'entered' &&
+    `
+    transform: scale(1);
+  `}
+
+  ${props =>
+    props.transitionstate === 'exiting' &&
+    `
+    transform: scale(0.5);
+  `}
+
+  ${props =>
+    props.transitionstate === 'exited' &&
+    `
+    transform: scale(0.5);
+  `}
+`
+
+const HeaderWrapper = styled.div`
+  flex-shrink: 0;
+`
+
+const FloatCloseButton = styled(CloseButton)`
+  position: absolute;
+  top: 16px;
+  right: 16px;
+  z-index: 2;
+
+  svg {
+    vertical-align: top;
+  }
+
+  &:focus {
+    outline: none;
+  }
+`
+
+const ContentWrapper = styled(Box)`
+  position: relative;
+  ${props => {
+    if (!props.enableoverflow) {
+      return `
+        overflow-y: scroll;
+        overflow-x: hidden;
+        height: 100%;
+        -webkit-overflow-scrolling: touch;
+      `
+    }
+  }}
+`
+
+const OverlayWrapper = styled.div`
+  display: table;
+  height: 100%;
+  width: 100%;
+`
+
+const DialogWrapper = styled(Box)`
+  display: table-cell;
+  vertical-align: middle;
+  ${props =>
+    props.enableoverflow &&
+    `
+    padding-top: 24px;
+    padding-bottom: 24px;
+  `};
+`
+
+const DialogInnerWrapper = styled(Flex)`
+  position: relative;
+  height: 100%;
+`
+
+const Modal = ({
+  isOpen,
+  onClose,
+  bg,
+  zIndex,
+  children,
+  imgMode,
+  width,
+  className,
+  header,
+  disableCloseButton,
+  enableOverflow,
+  height
+}) => (
+  <Transition in={isOpen} unmountOnExit timeout={500}>
+    {state => (
+      <Overlay onDismiss={onClose} zindex={zIndex} transitionstate={state}>
+        <OverlayWrapper>
+          <DialogWrapper enableoverflow={enableOverflow}>
+            <Dialog
+              width={width}
+              bg={bg}
+              height={enableOverflow ? null : height}
+              transitionstate={state}
+              className={className}
+              enableoverflow={enableOverflow ? 'true' : null}
+            >
+              <DialogInnerWrapper flexDirection="column">
+                {header && <HeaderWrapper>{header}</HeaderWrapper>}
+                <ContentWrapper
+                  p={imgMode ? 0 : 16}
+                  header={header}
+                  enableoverflow={enableOverflow}
+                >
+                  {enableOverflow && !disableCloseButton && (
+                    <FloatCloseButton
+                      data-testid="close"
```

Comment:
not a blocker but it would be good to namespace this test-id a bit more.  You never know if a component/host app that pulls this in and does a full render/mount test would be looking for something similar to this.

Reply:
will update